migrate_lifecycle.go

```diff
@@ -108,16 +104,16 @@ func (o *MigrationOptions) LeafHasher() func([]byte) ([][]byte, error) {
 // of the source tree and so no attempt is made to reject/deduplicate entries.
 func (o *MigrationOptions) WithAntispam(as Antispam) *MigrationOptions {
 	if as != nil {
-		o.followers = append(o.followers, func(ctx context.Context, lr LogReader) {
-			as.Populate(ctx, lr, o.bundleIDHasher)
-		})
+		o.followers = append(o.followers, as.Follower(o.bundleIDHasher))
```

Comment:
Can you add a comment to `bundleIDHasher` that this must always be immutable after struct initialization? An easy trap now would be to add a feature to allow that to become mutable, which would make this `WithAntispam` method behave differently depending on order of calls.